src/recaplog

```diff
@@ -20,145 +20,252 @@
+      if [[ -n ${empty_dirs} ]]; then
+        count_dirs=$( wc -l <<<"${empty_dirs}" )
+      fi
+      log INFO "${prefix_msg} ${count_dirs} empty directories."
+      for empty_dir in ${empty_dirs}; do
+        rmdir "${empty_dir}"
+      done
+    fi
+  fi
 }
 
-# cat tiny log files into daily log files
+# Function to compress/move daily log files
+# Previously the logs were concatenated and could be compressed, that is a
+# problem for parsing them at a later time as reported in:
+# https://github.com/rackerlabs/recap/pull/28
+# The new behaviour is to move the log files to a directory then opt for
+# compression, if compressed then it will delete the original logs.
 pack_old_logs() {
-	YESTERDAYDATE=$(date -d "now -1 day" "+%Y%m%d")
-	for item in $COMPRESS_ITEMS; do
-		print_msg DEBUG "packing ${item}..."
-		OUTPUTFILE=${BASEDIR}/${item}_daily_${YESTERDAYDATE}.log
-		#
-		if [ $LOG_COMPRESS -eq 1 ]; then
-			print_msg DEBUG "do compress old log files"
-			cat ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log | gzip > ${OUTPUTFILE}.gz
-			ERRCODE=$?
-			touch -t ${YESTERDAYDATE}0000 $OUTPUTFILE.gz
-		else
-			print_msg DEBUG "do not compress old log files"
-			cat ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log > $OUTPUTFILE
-			ERRCODE=$?
-			touch -t ${YESTERDAYDATE}0000 $OUTPUTFILE
-		fi
-		if [ $ERRCODE -eq 0 ]; then
-			for f in $(ls ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log | grep -v daily); do
-				print_msg DEBUG "deleting: $f"
-				rm -f $f
-			done 
-		else
-			print_msg ERROR "cat error, cannot delete old log files '"${BASEDIR}/${item}_${YESTERDAYDATE}-*.log"'"
-		fi
-	done
+  local -a errors
+  local YESTERDAYDATE=$( date -d "now -1 day" "+%Y%m%d" )
+  if [[ ${LOG_COMPRESS} -eq 1 ]]; then
+    log INFO "Compressing old log files"
+  else
+    log INFO "Not compressing old log files"
+  fi
+  # Backing up only items defined in the config
+  for item in ${BACKUP_ITEMS}; do
+    YESTERDAY_ITEM_DIR="${BASEDIR}/${item}_daily_${YESTERDAYDATE}"
+    err=''
+    log INFO "Packing ${item}..."
+    log_count=$(
+      {
+        ls -1 ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log | wc -l
+      } 2>&1; exit ${PIPESTATUS[0]}
+    )
+    if [[ $? -ne 0 ]]; then
+      log ERROR "An error ocurred while reading logs: ${log_count}, "\
+                "skipping..."
+      continue
+    fi
+    # Moving logs to yesterday item's directory
+    if [[ ${DRYRUN} ]]; then
+      log INFO "Would move ${log_count} logs to: ${YESTERDAY_ITEM_DIR}"
+    else
+      log INFO "Moving ${log_count} logs to: ${YESTERDAY_ITEM_DIR}"
+      err=$(
+        {
+          mkdir -p "${YESTERDAY_ITEM_DIR}";
+          mv ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log \
+             ${YESTERDAY_ITEM_DIR};
+        } 2>&1
+      )
+      if [[ ! -z ${err} ]]; then
+        log ERROR "An error ocurred while attempting to move logs ${err}"\
+                  ", skipping..."
+        continue
+      fi
+    fi
+    # Compressing logs
+    if [[ ${LOG_COMPRESS} -eq 1 ]]; then
```

Comment:
I'd suggest testing for "-ne 1" and jumping straight to 'continue', to reduce the levels of nesting.